OpenEdXMobile/src/main/java/org/edx/mobile/base/BaseFragmentActivity.java

```diff
@@ -279,6 +279,9 @@ protected boolean isLandscape() {
      */
     @SuppressWarnings("unused")
     public void onEvent(LogoutEvent event) {
+        environment.getRouter().forceLogout(this,
+                environment.getAnalyticsRegistry(),
+                environment.getNotificationDelegate());
         finish();
```

Comment:
IMO, we can remove the `finish()` from this method as we are clearing the stack in `forceLogout`

Reply:
Plus, in `environment.getRouter().forceLogout` we can remove the `EventBus.getDefault().post(new LogoutEvent());` as it can make a loop

Reply:
removed.

Reply:
added again the `finish();` statment for the safe side.